### PR TITLE
Fix(SVGParser) ignore missing xlink target issue on svg parsing

### PR DESCRIPTION
**Pull request by gloriousjob**
## Motivation

@closes  https://github.com/fabricjs/fabric.js/issues/9109

## Description

Sometimes an svg has use tags with an xlink with no target, which most parsers ignore. We should ignore these use tags as well.

## Changes

Adds a null check and returns early, preventing an NPE.
Also removed elementById since it was needed for IE8 support, which we no longer support (see support for everything else here: https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById).

### CODE REVIEW

**Review by ShaMan123 — COMMENTED** [review]
This looks good
Thanks!
Can this happen with other directives like clip paths? Should we igonre them as well?

**Comment by gloriousjob** [comment]
I didn't realize that was being used by the use tag (elements_parser apparently handles this). But I found an example and tested it. It worked normally and when I changed the id to a bad one, it ignored the clipping so looks like the code is already healthy for that.
 Only question I have now is how to get the 2 failures to pass. I'm not sure if this is because I'm using my own repo since I don't have write access on the official repo.

**Comment by ShaMan123** [comment]
> I didn't realize that was being used by the use tag (elements_parser apparently handles this). But I found an example and tested it. It worked normally and when I changed the id to a bad one, it ignored the clipping so looks like the code is already healthy for that.

Could you add a test?


>  Only question I have now is how to get the 2 check failures to pass. I'm not sure if this is because I'm using my own repo since I don't have write access on the official repo.

You are right, it is because of write access.

**Comment by gloriousjob** [comment]
> Could you add a test?

Tacked on additional test.

File: test/unit/parser.js [hunk 1]
```diff
@@ -804,6 +804,36 @@
...
+      `<svg viewBox="0 0 128 128" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+      <path id="heart" d="M10,30 A20,20,0,0,1,50,30 A20,20,0,0,1,90,30 Q90,60,50,90 Q10,60,10,30 Z" />
+      <use clip-path="url(#myClip)" href="#heart" fill="red" />
+      </svg>`;
+
+    fabric.loadSVGFromString(string).then(({ objects }) => {
+      assert.equal(objects[0].constructor.type, 'Path');
+      done();
+    });
+  });
```
Comment (asturur) [inline test/unit/parser.js:835]: please add those to jests instead of qunit.

**Comment by asturur** [comment]
Do we have an svg with missing xlink and can we see how the browser renders it?

File: test/unit/parser.js (hunk 1 above)
Comment (gloriousjob) [reply test/unit/parser.js]: done

**Comment by gloriousjob** [comment]
> Do we have an svg with missing xlink and can we see how the browser renders it?

The svg from the unit test looks like:


Pretty simple but without the fix, nothing renders in fabric whereas it renders like this with the fix.

**Comment by asturur** [comment]
i moved the read shape into the defs element so we can test that the use element with the wrong clip-path still work without the clippath.
If the path is outside defs, you never know if the parser ignored the wrong clip-path attribute or ignored the whole `<use>` node

**Review by asturur — APPROVED** [review]

**Comment by asturur** [comment]
@ShaMan123 i m merging this since is a good fix and we can push it up with the beta right away, ok?

**Comment by ShaMan123** [comment]
sure

**Pull request merged**